ethereum/core/src/main/java/org/hyperledger/besu/ethereum/mainnet/precompiles/privacy/PrivacyPrecompiledContract.java

```diff
@@ -160,8 +186,16 @@ public BytesValue compute(final BytesValue input, final MessageFrame messageFram
       privateStateUpdater.putTransactionStatus(
           txHash,
           BytesValue.of(
-              result.getStatus() == TransactionProcessor.Result.Status.SUCCESSFUL ? 1 : 0));
+              result.getStatus() == PrivateTransactionProcessor.Result.Status.SUCCESSFUL ? 1 : 0));
       privateStateUpdater.putTransactionResult(txHash, result.getOutput());
+
+      // TODO: this map could be passed through from @PrivacyBlockProcessor and saved once at the
```

Comment:
Are we doing this TODO or is it gonna be saved for later?

Reply:
suggestion for later - this is here to remember that we are accessing the `privacyGroupHeadBlockMap` once for every block and once for every transaction in that block. We can improve this by accessing only once every block and passing down the instance.

But I think we leave it like this.

Reply:
Question: how much effort it would be to change it to access once every block? Are we talking hours, days or weeks of work?